### PR TITLE
Link set first pass.

### PR DESCRIPTION
**Pull request by Nakort**

### CODE REVIEW

**Comment by Nakort** [comment]
Couple of things to think about, now that we are separating the inclusions/links logic, how far do we want to separate it? 

Do we want links to be built of the inclusions information, or should this information be added separately?
e.g: having an add_link macro for controllers. models.

File: lib/matterhorn/links/set_member.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+      def initialize(name, config, options={})
+        @name             = name
+        @config           = config
+        @context          = options[:context]
+        @url_type         = config.url_type
+        @relation_name    = options[:relation_name] || name
+        @associated_tense = test_singularity(name) ? :singular : :plural
+
+        if config.base.ancestors.include?(::Matterhorn::Base)
```
Comment (blasterpal) [inline lib/matterhorn/links/set_member.rb]: Isn't this duplicate of `construct_metatdata!` method?

Comment (Nakort) [inline lib/matterhorn/links/set_member.rb]: yes

**Comment by blasterpal** [comment]
My read on the spec is `links` are optional as it relates to `includes` but in this section, under "Attributes": 
http://jsonapi.org/format/#document-structure-resource-objects

it states that relations within a resource have to be represented as links. So I would personally opt for them to always exist even if inclusions represent them. It's simpler and a few more bytes won't matter.

**Pull request closed without merging**